src/runtime_src/core/pcie/driver/linux/xocl/subdev/clock.c

```diff
@@ -886,10 +890,12 @@ static int clock_ocl_freqscaling(struct clock *clock, bool force, int level)
 	for (i = 0; i < CLOCK_MAX_NUM_CLOCKS; i++)
 		curr[i] = clock_get_freq_impl(clock, i);
 
-	err = clock_freeze_axi_gate(clock, level);
+	if (freeze)
```

Comment:
please check the return value. if err==-ENODEV, means there is not axigate device. Then you do not have to add 'freeze' parameter.

Reply:
Hi @houlz0507 , we thought of adding that check, but its going to affect platforms which doesn't have axi-gate (ideally there should not be any platform). But, as we are at edge of PU1, just wanted to make sure we are not touching any other flows.

Reply:
ok. Please consider this in master branch.

Reply:
Yes @houlz0507 , we should make that change in master branch. Thanks

Reply:
I have taken care of this comment in master PR https://github.com/Xilinx/XRT/pull/4785